File: test1/step_by_step/averaging_and_stat/Step2_MNE_WB2spokes_connectivity_PSI_UVttest.m
clear
clc
close all
addpath('/home/rf02/rezvan/test1/BCT/2015_01_25 BCT')
in_path='/imaging/rf02/TypLexMEG/';

list_all = {'meg10_0378/101209/',
    'meg10_0390/101214/',
    'meg11_0026/110223/',
    'meg11_0050/110307/',
    'meg11_0052/110307/',
    'meg11_0069/110315/',
    'meg11_0086/110322/',
    'meg11_0091/110328/',
    'meg11_0096/110404/',
    'meg11_0101/110411/',
    'meg11_0102/110411/',
    'meg11_0112/110505/',
    'meg11_0104/110412/',
    'meg11_0118/110509/',
    'meg11_0131/110519/',
    'meg11_0144/110602/',
    'meg11_0147/110603/',
    };

fname_a='mean_coh_abstract_multitaper_8regions.mat';
fname_c='mean_coh_concrete_multitaper_8regions.mat';
Matx_c=zeros(8,8,2,2,17);
Matx_a=zeros(8,8,2,2,17);

fname=[in_path,list_all{1},fname_a];
load(fname)
% fname_s='plv_subtract_cwtmorlet.mat';
n_subj=length(list_all);
node_perc=[0.05:0.05:0.75];

tr=zeros(n_subj,1);
trb=zeros(n_subj,1);
%labellist = ['lh.posterior_fusiform','rh.posterior_fusiform', 'lh.lateraloccipital','rh.lateraloccipital', 'lh.ATLsmall','rh.ATLsmall', 'lh.entorhinal', 'rh.entorhinal']#labellist = [ 'leftATL-lh','rightATL-rh']#, 'medialtempright-rh','medialtempleft-lh']'lh.postcentral','rh.postcentral',

load('/imaging/rf02/TypLexMEG/icaanalysis_results/jpg/time_resolved/ROI_connectivity/graph/labels_46.mat')
load('/imaging/rf02/TypLexMEG/icaanalysis_results/jpg/time_resolved/ROI_connectivity/graph/node_angles46.mat')
close all
for cnt=1:17
    cnt
    fname=[in_path,list_all{cnt},fname_c];
    load(fname)
    Matx_c(:,:,:,1,cnt)=mean(con_cncrt(:,:,:,650:850),4);
    Matx_c(:,:,:,2,cnt)=mean(con_cncrt(:,:,:,750:950),4);
    
    fname=[in_path,list_all{cnt},fname_a];
    load(fname)
    Matx_a(:,:,:,1,cnt)=mean(con_abs(:,:,:,650:850),4);
    Matx_a(:,:,:,2,cnt)=mean(con_abs(:,:,:,750:950),4);
    %
    
end
for ii=1:size(Matx_a,5)%subj
    for jj=1:size(Matx_a,4)%time
        for kk=1:size(Matx_a,3)%bands
            for mm=1:size(Matx_a,2)
                for nn=1:size(Matx_a,1)
                    if mm>nn
                        Matx_a(nn,mm,kk,jj,ii)=Matx_a(mm,nn,kk,jj,ii);
                        Matx_c(nn,mm,kk,jj,ii)=Matx_c(mm,nn,kk,jj,ii);
                    end
                end
            end
        end
    end
end
ttest_a=zeros(8,8,2,2);
ttest_c=zeros(8,8,2,2);
ttest_s=zeros(8,8,4,2);
for jj=1:size(ttest_a,4)%time
    for kk=1:size(ttest_a,3)%bands
        for mm=1:size(ttest_a,2)
            for nn=1:size(ttest_a,1)
                [h,o,ci,stats]=ttest(squeeze(Matx_a(nn,mm,kk,jj,:)));
                ttest_a(nn,mm,kk,jj)=stats.tstat;
                [h,o,ci,stats]=ttest(squeeze(Matx_c(nn,mm,kk,jj,:)));
                ttest_c(nn,mm,kk,jj)=stats.tstat;
                [h,o,ci,stats]=ttest(squeeze(Matx_c(nn,mm,kk,jj,:)-Matx_a(nn,mm,kk,jj,:)));
                ttest_s(nn,mm,kk,jj)=stats.tstat;
            end
        end
    end
end

ttest_a(isnan(ttest_a))=0;
ttest_c(isnan(ttest_c))=0;
Ma=Matx_a;
Mc=Matx_c;


Matx_a(Matx_a>0)=1;
Matx_a(Matx_a<0)=0;
ma=mean(Matx_a,5);
Matx_c(Matx_c>0)=1;
Matx_c(Matx_c<0)=0;
mc=mean(Matx_c,5);
for ii=1:size(ma,3)%bands
    for jj=1:size(ma,4)%time
        figure, imagesc(squeeze(ttest_s(5:end,1:4,ii,jj)),[-2,2]), colorbar
        %figure,imagesc(squeeze(ttest_a(5:end,1:4,ii,jj)),[-2,2]), colorbar
    end
end

File: test1/step_by_step/averaging_and_stat/Step2_MNE_WB2spokes_connectivity_PSI_UVttest.py
"""
=========================================================================
Compute source space connectivity and visualize it using a circular graph
=========================================================================

This example computes the all-to-all connectivity between 68 regions in
source space based on dSPM inverse solutions and a FreeSurfer cortical
parcellation. The connectivity is visualized using a circular graph which
is ordered based on the locations of the regions.
"""

# Authors: Martin Luessi <mluessi@nmr.mgh.harvard.edu>
#          Alexandre Gramfort <alexandre.gramfort@telecom-paristech.fr>
#          Nicolas P. Rougier (graph code borrowed from his matplotlib gallery)
#
# License: BSD (3-clause)

print(__doc__)
import sys

#sys.path.append('/imaging/local/software/python_packages/scikit-learn/v0.14.1')
sys.path.append('/imaging/local/software/mne_python/latest_v0.9')

# for qsub
# (add ! here if needed) 
sys.path.append('/imaging/local/software/EPD/latest/x86_64/bin/python')
sys.path.insert(1,'/imaging/local/software/python_packages/scikit-learn/v0.16.1')

#sys.path.append('/imaging/local/software/mne_python/git-master-0.8/mne-python/')
###

import numpy as np
import mne
from mne.io import Raw
from mne.minimum_norm import (read_inverse_operator, apply_inverse, apply_inverse_epochs)
import scipy.io
from mne import find_events
from mne.connectivity import (spectral_connectivity, phase_slope_index)
import numpy as np
from mne import read_labels_from_annot
#import sklearn

data_path = '/imaging/rf02/TypLexMEG/' # root directory for your MEG data
subjects_dir = '/imaging/rf02/TypLexMEG/'    # where your MRI subdirectories are

event_path = event_path = '/group/erp/data/olaf.hauk/MEG/TypLexM/data/MF22_new/'   # where event files are

label_path = '/imaging/rf02/TypLexMEG/fsaverage/label/'

inv_path = '/imaging/rf02/TypLexMEG/'
inv_fname = 'InverseOperator_fft_1_48_clean_ica_EMEG-inv.fif'

# get indices for subjects to be processed from command line input
# 
print sys.argv
subject_inds = []
for ss in sys.argv[1:]:
   subject_inds.append( int( ss ) )



subject_inds=[0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16]#0,1,2,3,4,5,6,7,8]#,
print "subject_inds:"
print subject_inds
print "No rejection"

list_all = ['meg10_0378/101209/', 
'meg10_0390/101214/',
'meg11_0026/110223/', 
'meg11_0050/110307/', 
'meg11_0052/110307/', 
'meg11_0069/110315/', 
'meg11_0086/110322/', 
'meg11_0091/110328/', 
'meg11_0096/110404/', 
'meg11_0101/110411/', 
'meg11_0102/110411/', 
'meg11_0112/110505/',
'meg11_0104/110412/',  
'meg11_0118/110509/', 
'meg11_0131/110519/', 
'meg11_0144/110602/', 
'meg11_0147/110603/', 


]

# subjects names used for MRI data
subjects=['SS1_Meg10_0378',
'SS2_Meg10_0390',
'SS3_Meg11_0026',
'SS4_Meg11_0050',
'SS5_Meg11_0052',
'SS6_Meg11_0069',
'SS9_Meg11_0086',
'SS10_Meg11_0091',
'SS12_Meg11_0096',
'SS13_Meg11_0101',
'SS14_Meg11_0102',
'SS15_Meg11_0112',
'SS16_Meg11_0104',
'SS18_Meg11_0118',
'SS19_Meg11_0131',
'SS20_Meg11_0144',
'SS21_Meg11_0147'
]

ll = []
for ss in subject_inds:
 ll.append(list_all[ss])

print "ll:"
print ll

labels=range(8)
labellist = ['lh.posterior_fusiform','rh.posterior_fusiform', 'lh.lateraloccipital','rh.lateraloccipital', 'lh.ATLsmall','rh.ATLsmall', 'lh.entorhinal', 'rh.entorhinal']#labellist = [ 'leftATL-lh','rightATL-rh']#, 'medialtempright-rh','medialtempleft-lh']'lh.postcentral','rh.postcentral',
Matx_c=np.zeros((8,8,2,2,17))
Matx_a=np.zeros((8,8,2,2,17))

for ii,meg in enumerate(ll):        
    print meg
    path_c=data_path+ meg+'mean_coh_concrete_multitaper_8regions.mat'
    ltc=scipy.io.loadmat(path_c,mat_dtype=True); con_cncrt=ltc['con_cncrt']
    Matx_c[:,:,:,0,ii]=con_cncrt[:,:,:,650:850].mean(axis=3)
    Matx_c[:,:,:,1,ii]=con_cncrt[:,:,:,751:950].mean(axis=3)
    #scipy.io.savemat(path_c,{'label_mne_ctf_mn_ts_concrete_resampled':label_mne_ctf_mn_ts_concrete_resampled})
    path_a=data_path+ meg+'mean_coh_abstract_multitaper_8regions.mat'
    #scipy.io.savemat(path_a,{'label_mne_ctf_mn_ts_abstract_resampled':label_mne_ctf_mn_ts_abstract_resampled})
    lta=scipy.io.loadmat(path_a,mat_dtype=True); con_abs=lta['con_abs']
    Matx_a[:,:,:,0,ii]=con_abs[:,:,:,650:850].mean(axis=3)
    Matx_a[:,:,:,1,ii]=con_abs[:,:,:,751:950].mean(axis=3)
    print "lbl ts"